spec/routing/routes_spec.rb

```diff
@@ -0,0 +1,67 @@
+# referensed from bug664-1.patch
+# https://gist.github.com/benjaminvialle/4055208
+require 'spec_helper'
+
+describe 'Routing to main page', :type => :routing do
+  context 'Locale-less root' do
+    it 'routes / to login' do
+      expect(:get => "/").to route_to(
+        :controller => 'main',
+        :action => 'login'
+      )
+    end
+  end
+  
+  
+  context 'Root with locale' do
+    it 'routes /en/ to error' do
+      expect(:get => "/en/").not_to be_routable
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.